cmd/awscollector/Dockerfile

```diff
@@ -4,63 +4,45 @@
 ################################
```

Comment:
Any reasons to separate the Dockerfile and binaryBuild.Dockerfile thought? Also would be better to [update the doc ](https://github.com/aws-observability/aws-otel-collector/blob/main/docs/developers/build-docker.md)and notify we have supported bold amd64 and arm64

Reply:
The reason being that we have a docker file when we build the binary (we do not advise to use since it take a long time to build). The other docker file will copy the binary (should be used by the makefile and the ci).